.github/workflows/aws-deploy.yml

```diff
@@ -28,26 +28,42 @@ jobs:
   deploy:
     runs-on: ubuntu-latest
     steps:
+    - name: Extract branch name on push
+      if: github.event_name != 'pull_request'
+      shell: bash
+      run: echo "BRANCH_NAME=${GITHUB_REF#refs/heads/}"  >> $GITHUB_ENV
+      id: extract_branch
+
+    - name: Extract branch name on pull request
+      if: github.event_name == 'pull_request'
+      run: echo "BRANCH_NAME=${GITHUB_HEAD_REF}"  >> $GITHUB_ENV
+
+    - name: Print branch name
+      run: echo 'The branch name is' $BRANCH_NAME
+    
```

Comment:
Looks like you're using a similar answer from this StackOverflow post:
https://stackoverflow.com/questions/58033366/how-to-get-the-current-branch-within-github-actions

Why not use the first answer:
```
env:
 BRANCH_NAME: ${{ github.head_ref || github.ref_name }} 
```